packages/mui-material/src/TextField/TextField.js

```diff
@@ -142,6 +142,9 @@ const TextField = React.forwardRef(function TextField(inProps, ref) {
     // unset defaults from textbox inputs
     if (!SelectProps || !SelectProps.native) {
       InputMore.id = undefined;
+      if (!InputLabelProps?.component) {
+        InputLabelProps.component = 'span';
+      }
```

Comment:
@DarhkVoyd We shouldn't change this behavior at this time, as it would cause a breaking change

Instead, users who wish to get rid of the devtools error should use the [workaround](https://github.com/mui/material-ui/issues/38869#issuecomment-1792184844) in their applications instead

Reply:
@mj12albert I have removed the breaking change.